### PR TITLE
[jade] validate trajectory before execution

### PR DESCRIPTION
**Pull request by davetcoleman**
Discussion continued from https://github.com/ros-planning/moveit/pull/63#issuecomment-247538638

@rhaschke feel free to make changes to this shared branch as necessary

### CODE REVIEW

**Comment by v4hn** [comment]
As discussed in https://github.com/ros-planning/moveit/pull/63#issuecomment-241398750 ,
the idea was to remove the old constructor in kinetic (for safety reasons) but still provide it in indigo.
Because jade has been officially released in the meanwhile, we should also provide the old constructor in jade. I would propose to deprecate the old constructor in jade.

**Comment by davetcoleman** [comment]
I agree with @v4hn

**Comment by rhaschke** [comment]
OK, I will prepare this.

**Comment by rhaschke** [comment]
Why not deprecate the old constructor in Indigo too? This is a security feature and we should enforce/motivate people to switch asap.

**Comment by davetcoleman** [comment]
If the deprecation is simple a warning during compile time, +1 to adding it to indigo too

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 1]
```diff
@@ -872,7 +885,7 @@ bool TrajectoryExecutionManager::distributeTrajectory(const moveit_msgs::RobotTr
 
 bool TrajectoryExecutionManager::validate(const TrajectoryExecutionContext &context) const
 {
-  if (allowed_start_tolerance_ == 0) // skip validation on this magic number
+  if (!csm_ || allowed_start_tolerance_ == 0) // skip validation on this magic number
```
Comment (davetcoleman) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: improve comment for new condition

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 2]
```diff
@@ -171,6 +181,9 @@ void TrajectoryExecutionManager::initialize()
 
   reconfigure_impl_ = new DynamicReconfigureImpl(this);
 
+  if (!csm_)
+    ROS_ERROR_NAMED("traj_execution","Trajectory validation before execution is disabled, because no CurrentStateMonitor was provided.");
```
Comment (davetcoleman) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: This sentence was hard to read - remove the "before execution" part

Also, I think this should just be a warning

**Comment by rhaschke** [comment]
@davetcoleman I addressed your comments. According to https://github.com/ros-planning/moveit/pull/225#issuecomment-247705788 this could be cherry-picked to Indigo 1:1.

**Comment by davetcoleman** [comment]
+1

**Comment by rhaschke** [comment]
@davetcoleman Any reason to not merge yet?

**Comment by davetcoleman** [comment]
feel free to merge (my pr) and cherry-pick to indigo

**Comment by rhaschke** [comment]
Cherry-picked into Indigo.

**Pull request merged**